src/backends/torch/torchlib.cc

```diff
@@ -212,37 +250,48 @@ namespace dd
       torch::load(_classif, model._weights, _device);
     if (!model._proto.empty())
       {
-        _native = std::make_shared<CaffeToTorch>(model._proto);
+        _graph = std::make_shared<CaffeToTorch>(model._proto);
         if (!model._traced.empty())
-          torch::load(_native, model._traced, _device);
+          torch::load(_graph, model._traced, _device);
+      }
+    if (!model._native.empty())
+      {
+        std::shared_ptr<NativeModule> m;
```

Comment:
My understanding is that at this stage, the native template hasn't been set. Before loading the model, we may want to put down a call to `NativeFactory::from_template` ?

Reply:
in case of native module/template, torchmodule::load should be called after factory::from_template, and it seems to me that it is the case in the code, i am not sure to understand your point

Reply:
When `TorchModule::load` is called from `init_mllib`, it doesn't seem that at that stage `from_template` has been called. Actually since one call is mandatory before the other, we may want to regroup them into a function ?

Reply:
in case of nbeats, it is not possible to instanciate model before having exact data size (length of timeseries), ie before the transform, so `init_mllib` cannot instanciate using `NativeFactory::from_template` . init_mllib loads a native net if available in repo, but it will be replaced by `NativeFactory::from_template` followed by another `TorchModule::load` at train or test time .
All we can do is remove this first model loading during init_mllib